packages/block-editor/src/store/reducer.js

```diff
@@ -1737,11 +1737,17 @@ export function highlightedBlock( state, action ) {
  */
 export function lastBlockInserted( state = {}, action ) {
 	switch ( action.type ) {
-		case 'ADD_LAST_BLOCK_INSERTED':
-			return { ...state, clientId: action.clientId };
+		case 'INSERT_BLOCKS':
+			if ( ! action.updateSelection || ! action.blocks.length ) {
+				return state;
+			}
 
-		case 'CLEAR_LAST_BLOCK_INSERTED':
-			return {};
+			const clientId = action.blocks[ 0 ].clientId;
+			const source = action.meta?.source;
+
+			return { clientId, source };
+		case 'RESET_BLOCKS':
```

Comment:
`RESET_BLOCKS` action is triggered in different situations like for example when executing undo/redo actions. Since it's resetting the blocks, the last block inserted should be cleared, this would be similar to the `CLEAR_LAST_BLOCK_INSERTED` action.

Reply:
Understood. This is a more optimal approach that I prefer since it relies on an existing action that will allow us to emulate the `CLEAR_LAST_BLOCK_INSERTED` behavior. The less code the better 😄